tests/test_data_utils.cpp

```diff
@@ -149,122 +253,177 @@ void GenerateGpsVel(test_message_t &msg, gps_vel_t &gps, int i, float f, bool in
     msg.data.gpsVel = gps;
     msg.dataHdr.id = DID_GPS1_VEL;
     msg.dataHdr.size = sizeof(gps_vel_t);
+    return true;
 }
 
-void GenerateISB(test_message_t &msg, int i, float f)
+bool GenerateISB(test_message_t &msg, int i, float f)
 {
     static bool init = true;
-    static pimu_t pimu = {};
-    static ins_1_t ins1 = {};
-    static gps_pos_t gpsPos = {};
-    static gps_vel_t gpsVel = {};
 
     msg.ptype = _PTYPE_INERTIAL_SENSE_DATA;
 
     // Initialize data
     if (init)
     {
         init = false;
-        GeneratePimu(  msg, pimu,   i, f, true);
-        GenerateIns1(  msg, ins1,   i, f, true);
-        GenerateGpsPos(msg, gpsPos, i, f, true);
-        GenerateGpsVel(msg, gpsVel, i, f, true);
+        
+        CurrentGpsTimeMs(s_gpsTowOffsetMs, s_gpsWeek);
+        s_timeMs = 0;
+        GeneratePimu(  msg, s_pimu,   i, f, true);
+        GenerateIns1(  msg, s_ins1,   i, f, true);
+        GenerateGpsPos(msg, s_gpsPos, i, f, true);
+        GenerateGpsVel(msg, s_gpsVel, i, f, true);
     }
 
-    switch (i%4)
-    {
-    default:    GeneratePimu(  msg, pimu,   i, f);   break;
-    case 1:     GenerateIns1(  msg, ins1,   i, f);   break;
-    case 2:     GenerateGpsPos(msg, gpsPos, i, f);   break;
-    case 3:     GenerateGpsVel(msg, gpsVel, i, f);   break;
+    if (GeneratePimu(  msg, s_pimu,   i, f)) { return true; }
+    if (GenerateIns1(  msg, s_ins1,   i, f)) { return true; }
+    if (GenerateGpsPos(msg, s_gpsPos, i, f)) { return true; }
+    if (GenerateGpsVel(msg, s_gpsVel, i, f)) { return true; }
+
+    return false;
+}
+
+bool timeIsSameAndSet(uint32_t &msgMs, uint32_t timeMs)
+{
+    if (msgMs != timeMs)
+    {   
+        msgMs = timeMs;
+        return true;
     }
+
+    return false;
 }
 
-void GenerateUblox(test_message_t &msg, int i, float f)
+bool GenerateNMEA(test_message_t &msg, int i, float f)
 {
-    (void)f;
+    // if (timeIsSameAndSet(s_msgTimeMs.nmeaPImu, s_msgTimeMs.pimu))
```

Comment:
Do we need this dead code?   If it needs keeping, it should include a comment explaining why its still here.  Otherwise, if it wasn't good enough to keep around the first time... let's kill it (again)!

Reply:
Yes, we do want that code.  Thanks.